### DISCUSSION TITLE
Allow simple maths expressions in the Ingridients editor > Quanitity

### DISCUSSION THREAD

**Discussion by BlenderDefender** [mealie-recipes/mealie#3183] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

When editing recipes, I'd often like to use maths expressions like addition, multiplication, division or subtraction for updating the quantity of an ingredient.
Two scenarios I've had, where this would be helpful:
1. I've tried out a recipe and now I'm updating it according to my experience. I've used about 50 grams more of one ingredient and only 90% of another ingredient
2. Scaling a recipe down after scraping it. A source defaults to a recipe yield of 6 servings, I'd like to scale it down to 2 servings. Sure, I could use the scale down factor in the recipe display page, but I'd like to have it saved as a scaled down version.

### Please provide a concise description of the feature that would resolve your issue.

Allowing simple maths expressions in the quantity field, using a library like `math-expression-evaluator`, would resolve my issue.
[Feature Request Mockup.webm](https://github.com/mealie-recipes/mealie/assets/61777131/3a6f46fd-cb66-476c-b464-f32afff1833d)

### Please consider and list out some caveats or tradeoffs made in your design decision

- From what I've found out, having a type "number" input doesn't work with this feature, so the quantity input would be turned into a regular text field
- This would add another dependency (`math-expression-evaluator`: 68 kB) to Mealie, unless someone is willing to write a maths parser for this feature request.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by indykoning** [comment]
Personally i am **very** interested in scenario 2.
I live alone and sometimes with 2 people.
Mostly all recipes are made for 4 or 6 people. Scaling down is possible but i have to do it every time, and i understand how it works. But my partner does not.

Even if it's only the scaling option being available during editing the ingredients i'll be happy as i'll be able to scale there, and setting the servings when viewing the recipe will make sense

**Comment by irnutsmurt** [comment]
I'd gladly drop a few dollars to implement this feature so I can adjust the recipe portions on the fly like doubling, or halving the servings.

**Reply by michael-genson** [reply]
You can adjust the entire recipe on the fly if your ingredients are parsed into quantities

**Comment by ChipSkylark37** [comment]
EDIT:
I tested doing a 'Serves 12, Yields 1' instead of 'Services 1, Yields 12' as a workaround for a cupcake recipe, which works OK, but not ideal. Since I would typically consider a serving of a baked item like that as the base, and the yield as number of cupcakes, cookies, slices of cake, etc. In this instance, if I really did want to make '5 cupcakes' for some reason, this method works better since I can use the + and - to get a precise amount vs having to try and guess the decimal equivalent. For something like cake though this wouldn't work since if I wanted '2 slices of cake' I can't easily scale that down, I'd still have to make the whole thing, or find a different baking method. 

=================

I currently parse the recipe down, using the 'scaling' feature. Though you cannot seemingly scale down below 1 via the + and - buttons, you can manually do it. So if a cupcake recipe 'serving' yields 12 cupcakes, I can adjust the scaling to be .25 or something else to yield a lower number. 

However, it doesn't appear to save that scale, so upon refresh it goes back to the original. Perhaps a 'save as default service / yield' or something? If I am testing a recipe or just making a small batch for myself (more often the case than making a full batch of 12). I know I could adjust the recipe manually to account for a serving to be 3 and then SCALE UP if desired, but a few 'presets' would also be nice. 

What if you allowed us to enter the desired YIELD and it adjusted the scaling automatically? "This recipe has a yield of 12 cupcakes" and I select "I want 3" and it adjusted based off that. Just thinking through some various ways to get to the same sort of outcome. (Made EDIT above, but this wouldn't cover all scenarios)

Thought separate, another 'usability' piece is when entering cook mode, you get the 'serves XY' (if I changed it from 1 serving to .25) but you don't get the YIELD in the cook mode. So I can see I scaled it down, but the original yield of 12 cupcakes into 3 cupcakes isn't shown. Could that easily be added?
